diceRoll/diceRollUnitTests.js

```diff
@@ -26,15 +25,22 @@ function runTests(runTF) {
 
 function UnitTest(testName, functionToBeTested, expectedOutput) {
   this.report;
-  this.functionToBeTested = testName;
-  this.functionReturned = functionToBeTested;
-  if (this.functionReturned == expectedOutput) {
-    this.report = this.functionToBeTested + '\n\tPassed.\n\tReturned: ' + this.functionReturned;
-    console.log(this.report);
-    return this.report;
-  } else {
-    this.report = this.functionToBeTested + '\n\tFailed.\n\tReturned: ' + this.functionReturned;
-    console.log(this.report);
-    return this.report;
+  this.testName = testName;
+  this.functionToBeTested = functionToBeTested;
+  this.functionReturned = this.functionToBeTested();
+  this.expectedOutput = expectedOutput;
+  if (typeof this.expectedOutput === 'function') this.expectedOutput = this.expectedOutput();
```

Comment:
actually, you need to execute the expectedOutput inside the try statement, otherwise there is no reason to use a try/catch.

```
if (typeof this.expectedOutput === 'function') {
  try {
      this.expectedOutput = this.expectedOutput();
      if (this.functionReturned != this.expectedOutput) {
       this.report = '\n\tFailed.\n\tReturned: ' + this.functionReturned + '\n\tExpected: ' + this.expectedOutput;
       throw this.report;
     }
   } catch (error) {
     console.log(this.testName + error);
     return this.testName + this.report;
   }
}
```